src/GroupDetails.ts

```diff
@@ -0,0 +1,63 @@
+import { GroupHeading } from './Group';
+import type { Task } from './Task';
+
+export class IntermediateTaskGroupsStorage extends Map<string[], Task[]> {}
+
+export class IntermediateTaskGroups {
+    public groups = new IntermediateTaskGroupsStorage();
+
+    addTask(keys: string[], task: Task) {
+        const groupForKey = this.getOrCreateGroupForKey(keys);
+        groupForKey?.push(task);
+    }
+
+    private getOrCreateGroupForKey(newKey: string[]) {
+        for (const [key, taskGroup] of this.groups) {
+            // TODO Review for efficiency
+            if (JSON.stringify(key) == JSON.stringify(newKey)) {
```

Comment:
What exactly are you trying to achieve here? Do you want to compare two arrays?

Should it be `===` instead of just two equal signs?

Reply:
This is another of the areas where I need to add comments to clarify the implementation...
Am hoping someone will then be able to come up with a clearer implementation.

Reply:
I'm hoping this is clearer now, both in the code, and in the long description comment in the same file.